strangerstudios/paid-memberships-pro
# Checking if there is a discount code before doing these checks to avo…

### All Submissions:






### Changes proposed in this Pull Request:



I had noticed while running PMPro Toolkit's checkout debug emails that for free checkouts, I would always get an email with pmpro_error set to "No code was given to check."

When I dug into the code, it seems that we are calling the pmpro_checkDiscountCode() function even in cases where there is no code set.

As far as I know, there is no reason this is needed. So I added a check that ! empty( $discount_code ) before making those calls. We set $discount_code_id to "" if there is no discount code to make sure that var is set/clear when updating the order later on.

### How to test the changes in this Pull Request:

1. Enable the PMPro Toolkit plugin and enable the checkout tracking for errors only.
2. Checkout for a free level
3. Fill out all fields and leave the discount code field blank
4. Click submit.
5. Note that no checkout error email is generated.

### Other information:




### Changelog entry

> BUG FIX/ENHANCEMENT: Fixed issue where a "No code was given to check" error was showing up in checkout debug emails for all checkouts.
